docs/loading/Flink-connector-starrocks.md

```diff
@@ -1,173 +1,647 @@
+(
+    `id` int(11) NOT NULL COMMENT "",
+    `name` varchar(65533) NULL DEFAULT "" COMMENT "",
+    `score` int(11) NOT NULL DEFAULT "0" COMMENT ""
+)
+ENGINE=OLAP
+PRIMARY KEY(`id`)
+COMMENT "OLAP"
+DISTRIBUTED BY HASH(`id`);
+```
+
+#### Set up Flink environment
+
+- Download Flink binary [Flink 1.15.2](https://archive.apache.org/dist/flink/flink-1.15.2/flink-1.15.2-bin-scala_2.12.tgz), and unzip it to directory `flink-1.15.2`.
+- Download [connector 1.2.7](https://repo1.maven.org/maven2/com/starrocks/flink-connector-starrocks/1.2.7_flink-1.15/flink-connector-starrocks-1.2.7_flink-1.15.jar), and put it into the directory `flink-1.15.2/lib`.
+- Run the following commands to start a Flink cluster:
+
+    ```shell
+    cd flink-1.15.2
+    ./bin/start-cluster.sh
+    ```
+
+### Run with Flink SQL
+
+- Run the following command to start a Flink SQL client.
+
+    ```shell
+    ./bin/sql-client.sh
+    ```
+
+- Create a Flink table `score_board`, and insert values into the table via Flink SQL Client.
+Note you must define the primary key in the Flink DDL if load to a StarRocks primary key table. It's optional for other types of StarRocks table.
+
+    ```SQL
+    CREATE TABLE `score_board` (
+        `id` INT,
+        `name` STRING,
+        `score` INT,
+        PRIMARY KEY (id) NOT ENFORCED
+    ) WITH (
+        'connector' = 'starrocks',
+        'jdbc-url' = 'jdbc:mysql://127.0.0.1:9030',
+        'load-url' = '127.0.0.1:8030',
+        'database-name' = 'test',
+        'table-name' = 'score_board',
+        'username' = 'root',
+        'password' = ''
+    );
+
+    INSERT INTO `score_board` VALUES (1, 'starrocks', 100), (2, 'flink', 100);
+    ```
+
+### Run with Flink DataStream
+
+There are several ways to implement a Flink DataStream job according to the type of the input records, such as a csv Java `String`, a json Java `String` or a custom Java object.
+
+- The input records are csv-format `String`. See [LoadCsvRecords](https://github.com/StarRocks/starrocks-connector-for-apache-flink/tree/main/examples/src/main/java/com/starrocks/connector/flink/examples/datastream/LoadCsvRecords.java) for a complete example.
+
+    ```java
+    // Generate csv-format records. Each record has three fields separated by "\t". These
+    // fields correspond to the columns `id`, `name`, and `score` in StarRocks table.
+    String[] records = new String[]{
+            "1\tstarrocks-csv\t100",
+            "2\tflink-csv\t100"
+    };
+    DataStream<String> source = env.fromElements(records);
+
+    // Configure the connector with the required properties, and you also need to add properties
+    // "sink.properties.format" and "sink.properties.column_separator" to tell the connector the
+    // input records are csv-format, and the separator is "\t". You can also use other separators
+    // in the records, but remember to modify the "sink.properties.column_separator" correspondingly
+    StarRocksSinkOptions options = StarRocksSinkOptions.builder()
```

Comment:
![image](https://github.com/StarRocks/starrocks/assets/57167462/1a334650-dad7-4d41-932a-e96768c47261)
there are many (not just the ones in the above figure) comments in code snippets, and these comments seem to use bad wording. I suggest optimizing all of them at least make them readable.